api/src/index.node.ts

```diff
@@ -44,11 +46,22 @@ console.log(`FPX Server is running: http://localhost:${port}`);
 const serviceTargetArgument = process.env.FPX_SERVICE_TARGET;
 const probeMaxRetries = 10;
 const probeDelay = 1000;
-probeRoutesWithExponentialBackoff(
-  serviceTargetArgument,
-  probeMaxRetries,
-  probeDelay,
-);
+const watchDir = process.env.FPX_WATCH_DIR ?? process.cwd();
+
+const ignoredPaths = getIgnoredPaths();
+
+fs.watch(watchDir, { recursive: true }, async (eventType, filename) => {
+  if (!filename) return
+  if (ignoredPaths.includes(filename)) return
```

Comment:
I guess this will still use ignored files if their parent folder is only declared in the ignored path?
Something like `target` where the filename could me `target/file.*`. Maybe we should do glob matching here?

I know there's only hardcoded values right now but this could potentially bite us in the rear later on

Reply:
yea I was torn how far to go with it for this initial implementation. To some extent the cost of the probe restarting more often than perhaps it should is minimal so I didn't bother with a more comprehensive ignorePath resolution. At the same time I'm open to making this more robust later on.

Reply:
Yeah would agree, maybe a ticket or a comment should be fine for now :)

Reply:
i ended up implementing glob matching, since we were effectively watching all of `.wrangler`, which got quite noisy and i thiiink was causing some strange errors with wrangler itself?